index.bs

```diff
@@ -5115,7 +5131,7 @@ partial interface MLGraphBuilder {
     1. If [=MLGraphBuilder/validating operand=] with [=this=] and |input| returns false, then [=exception/throw=] a {{TypeError}}.
     1. Let |outputShape| be an empty array of {{unsigned long}}.
     1. If |newShape|'s [=list/size=] is 0, set |outputShape| to an empty [=/list=] for a scalar.
-    1. If any value in |newShape| is 0, then [=exception/throw=] a {{TypeError}}.
+    1. If any [=list/item=] in |newShape| is 0 or is not a [=valid dimension=], then [=exception/throw=] a {{TypeError}}.
```

Comment:
Nit: "valid dimension" already ensures the dimension is greater than 0.

```suggestion
    1. If any [=list/item=] in |newShape| is not a [=valid dimension=], then [=exception/throw=] a {{TypeError}}.
```

Reply:
I left this alone because if #391 allows 0 widely we might not want it for reshape... but on further thought that's caught by the later steps (_"If product of all values in newShape is not equal to inputElementCount, then throw a TypeError."_) So applying!